File: LeetCode/Python/#441 Arraning Coins.py
class Solution(object):
    def arrangeCoins(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n == 1:
            return 1
        res = 0
        tmp = 1
        n = n - tmp
        while n>=0:
            res += 1
            tmp += 1
            n -= tmp

        return res

File: LeetCode/JavaScript/#441 Arraning Coins.js
/**
 * @param {number} n
 * @return {number}
 */

var arrangeCoins = function(n) {
    var v = Math.sqrt(2 * n + 1/4);
    var k = (v - 0.5) | 0;
    return k;
};

// Second 
var arrangeCoins = function(n) {
    if(n===1)return 1;
    var res = 0;
    var tmp = 1;
    n = n - tmp;
    while(n>=0){
      res++;
      tmp++;
      n -= tmp;
    }
    return res;
};